pipeline_dp/dataframes.py

```diff
@@ -309,50 +320,128 @@ def count(self, name: str = None) -> 'QueryBuilder':
 
     def build_query(self) -> Query:
         """Builds the DP query."""
         if self._by is None:
             raise NotImplementedError(
-                "Global aggregations are not implemented yet. Call groupby.")
-        if not self._metrics:
+                "Global aggregations are not implemented yet. Call groupby")
+        # Validation.
+        if not self._aggregations_specs:
             raise ValueError(
-                "No aggregations in the query. Call for example count.")
-        return Query(
-            self._df,
-            Columns(self._privacy_unit_column, self._by, self._value_column),
-            self._metrics, self._contribution_bounds, self._public_keys)
+                "No aggregations in the query. Call count, sum, mean etc")
+        # 1. Not more than 1 value column
+        input_columns = [
+            spec.input_column
+            for spec in self._aggregations_specs
+            if spec.input_column is not None
+        ]
+        if len(set(input_columns)) > 1:
+            raise NotImplementedError(
+                f"Aggregation of only one column is supported, but {input_columns} given"
+            )
+        input_column = input_columns[0] if input_columns else None
+        # 2. Each metric was added only once
+        metrics = [spec.metric for spec in self._aggregations_specs]
+        if len(set(metrics)) != len(metrics):
+            raise ValueError("Each aggregation can be added only once.")
+        # 3. Value caps are given if needed and consistent
+        min_value, max_value = self._get_value_caps()
+
+        # Create Contribution bounds
+        contribution_bounds = ContributionBounds(
+            max_partitions_contributed=self._max_partitions_contributed,
+            max_contributions_per_partition=self.
+            _max_contributions_per_partition,
+            min_value=min_value,
+            max_value=max_value)
+
+        metric_to_output_column = dict((spec.metric, spec.output_column)
+                                       for spec in self._aggregations_specs)
+
+        return Query(self._df,
+                     Columns(self._privacy_unit_column, self._by,
+                             input_column), metric_to_output_column,
+                     contribution_bounds, self._public_keys)
+
+    def _add_aggregation(self,
+                         aggregation_spec: _AggregationSpec) -> 'QueryBuilder':
+        if self._by is None:
+            raise ValueError(
+                "Global aggregations are not supported. Use groupby")
+        self._aggregations_specs.append(aggregation_spec)
+        return self
+
+    def _get_value_caps(self) -> Tuple[Optional[float], Optional[float]]:
+        metrics = set([spec.metric for spec in self._aggregations_specs])
+        # COUNT, PPRIVACY_ID_COUNT do not require caps.
+        metrics_which_need_caps = metrics.difference(
+            [pipeline_dp.Metrics.COUNT, pipeline_dp.Metrics.PRIVACY_ID_COUNT])
```

Comment:
It's a good point that we can put the logic checking if needed caps in some other place. Having them as a part of `Metric` object is not nice, because it's actually implementation dependendant (for example in future we can implement algorithm which finds caps automatically).
Another option is to put it in some helper function closer to `Metric`. Thoug, I'm not sure. So let's leave as is for now.